weblate/screenshots/tests.py

```diff
@@ -245,3 +248,43 @@ def test_translation_manipulations(self):
             {"source": source_pk},
         )
         self.assertEqual(screenshot.units.count(), 0)
+
+
+class ScreenshotVCSTest(RepoTestCase):
+    """Test class for syncing vcs screenshots in weblate."""
+
+    def setUp(self):
+        super().setUp()
+        self.component = self._create_component(
+            "json",
+            "intermediate/*.json",
+            screenshot_filemask="*.png",
+        )
+
+    def test_update_screenshots_from_repo(self):
+        repository = self.component.repository
+        last_revision = repository.last_revision
+
+        with open(TEST_SCREENSHOT, "rb") as handle, open(
+            os.path.join(repository.path, "test.png"), "wb"
+        ) as f:
+            f.write(handle.read())
+
+        with repository.lock:
+            repository.set_committer("Second Bar", "second@example.net")
+            filenames = ["test.png"]
+            repository.commit(
+                "Test commit", "Foo Bar <foo@bar.com>", timezone.now(), filenames
+            )
+            self.component.trigger_post_update(last_revision, skip_push=True)
+
+        # Verify that screenshot has been added after the signal.
+        self.assertEqual(
+            len(
+                Screenshot.objects.filter(
+                    translation__component=self.component,
+                    repository_filename="test.png",
+                )
+            ),
```

Comment:
Using `count()` on queryset is faster because it does count on the database side.

```suggestion
            Screenshot.objects.filter(
                translation__component=self.component,
                repository_filename="test.png",
            ).count(),
```